packages/flame/lib/src/components/base_component.dart

```diff
@@ -143,7 +143,10 @@ abstract class BaseComponent extends Component {
   /// For children that don't need preparation from the game instance can
   /// disregard both the options given above.
   Future<void> addChild(Component child, {Game? gameRef}) async {
-    gameRef ??= (this as HasGameRef).gameRef;
+    if (this is HasGameRef) {
+      final c = this as HasGameRef;
+      gameRef ??= c.hasGameRef ? c.gameRef : null;
```

Comment:
We should probably throw here is the a child is added that has `hasGameRef` but then `gameRef` and `c.gameRef` is null?

Reply:
Oh right, we will throw, in the getter.